syntax = "proto3";

package dblib;

// Result message for structured logging
message Result {
  // Unique identifier for the run
  string run_id = 1;

  // Random seed used for the run
  int32 random_seed = 2;

  // Incremental iteration number within the run
  int32 iteration_number = 3;
  
  // Type of database operation being benchmarked
  OpType op_type = 4;
  
  // Initial size of the database in bytes (or rows, depending on context)
  int64 initial_db_size = 5;
  
  // Name of the table being operated on
  string table_name = 6;
  
  // Schema of the table (DDL or schema description)
  string table_schema = 7;
  
  // Number of keys touched during the operation
  int64 num_keys_touched = 8;

  // Latency for operations (in seconds)
  double latency = 9;

  // Disk size of the database before the operation (in bytes)
  int64 disk_size_before = 10;

  // Disk size of the database after the operation (in bytes)
  int64 disk_size_after = 11;

  // The SQL query that was executed
  string sql_query = 12;
}

// Enum for different types of database operations
enum OpType {
  UNSPECIFIED = 0;
  BRANCH_CREATE = 1;
  BRANCH_CONNECT = 2;
  READ = 3;
  INSERT = 4;
  UPDATE = 5;
  COMMIT = 6;
}

